06.bowling_object/lib/game.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require_relative './frame'
+require_relative './last_frame'
+require_relative './shot'
+
+class Game
+  def initialize
+    @frames = Array.new(10) do |index|
+      index == 9 ? LastFrame.new : Frame.new
+    end
+  end
+
+  def play(marks)
+    index = 0
+    @frames.each do |frame|
+      loop do
+        break if frame.full?
+
+        shot = Shot.new(marks[index])
+        frame.set(shot)
+        index += 1
+      end
+    end
```

Comment:
```suggestion
    copied_marks = marks.dup
    @frames.each do |frame|
      until frame.full?
        shot = Shot.new(marks.shift)
        frame.set(shot)
      end
    end
```

みたいにするとシンプルかも。

破壊的メソッドはなるべく避けた方がいいですが、対象がローカル変数で破壊的メソッドの方がシンプルになる場合は例外的に使っても良いと思います